lymph/unilateral.py

```diff
@@ -165,13 +167,27 @@ def _edge_lookup(self):
     @property
     def allowed_lnl_states(self) -> List[int]:
         """Return the list of allowed states for the LNLs."""
-        return self. lnls[0].allowed_states
+        return self.lnls[0].allowed_states
 
 
     def __str__(self) -> str:
         """Print info about the instance."""
         return f"Unilateral with {len(self.tumors)} tumors and {len(self.lnls)} LNLs"
 
+    #here you might want to change the type of output
+    def print_graph(self):
+        'produces a mermaid graph of the given model'
+        graph = ('flowchart TD\n')
+        for index, node in enumerate(self.nodes):
+            for edge in self.nodes[index].out:
+                line = f"{node.name} -->|{edge.spread_prob}| {edge.end.name} \n"
+                graph += line
+        graphbytes = graph.encode("ascii")
+        base64_bytes = base64.b64encode(graphbytes)
+        base64_string = base64_bytes.decode("ascii")
+        display(Image(url="https://mermaid.ink/img/" + base64_string))

```

Comment:
Cool, I didn't know that was possible 😅

But to be honest, I don't really want this library to depend on `IPython`. I think it would be best to have the function `print_graph` be capable of returning the string that represents the mermaid graph (so that someone can put it in their markdown readme, for example) and the URL that you pass to the `Image` constructor here, depending on the user's preference.